java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -36,35 +38,43 @@
 import java.util.stream.Stream;
 
 public class MethodYield {
+  private final boolean varArgs;
   Constraint[] parametersConstraints;
   int resultIndex;
   @Nullable
   Constraint resultConstraint;
   boolean exception;
 
-  public MethodYield(int arity) {
+  public MethodYield(int arity, boolean varArgs) {
     this.parametersConstraints = new Constraint[arity];
+    this.varArgs = varArgs;
     this.resultIndex = -1;
     this.resultConstraint = null;
     this.exception = false;
   }
 
+
   @Override
   public String toString() {
     return "{params: " + Arrays.toString(parametersConstraints) + ", result: " + resultConstraint + " (" + resultIndex + "), exceptional: " + exception + "}";
   }
 
-  public Collection<ProgramState> statesAfterInvocation(List<SymbolicValue> invocationArguments, ProgramState programState, Supplier<SymbolicValue> svSupplier) {
+  public Collection<ProgramState> statesAfterInvocation(List<SymbolicValue> invocationArguments, List<Type> invocationTypes, ProgramState programState,
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "statesAfterInvocation" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)